public/javascripts/controllers/buyer-ctrl.js

```diff
@@ -227,6 +227,89 @@ angular.module('jasmic.controllers')
                 $scope.demand.de_until = moment().add(7, 'days').toDate();
             };
 
+            $scope.editDemandForm = function (obj) {
+
+                $scope.edit_demand = !$scope.edit_demand;
+
+                $scope.demand = obj;
+
+                //for some reason the dates needed to be passed through MomentJS as seen below to be assigned to the form inputs
```

Comment:
The obj parameter does contain the corresponding dates along with all the other information for the demand but the date fields in the form don't handle them correctly unless I pass them through the momentJS library. In other words there seems to be a disconnect between the date format returned from the mongo database and the date input field (which I used the momentJS library to resolve).

Outputting the dates via the console before and after passing them through MomentJS should illustrate what I mean if you're curious to see the exact formatting differences.